### PR TITLE
feat: add getOrDefault to Map

### PR DESCRIPTION
**Pull request by yjhmelody**
Add `getOrDefault` to prevent panic when get value of map.

- [ x ] I've read the contributing guidelines
- [ x ] I've added my name and email to the NOTICE file

### CODE REVIEW

**Comment by yjhmelody** [comment]
I do not know to update the fixture

File: tests/compiler/std/math.release.wat [hunk 1]
```diff
@@ -49900,7 +49900,7 @@
    call $~lib/builtins/abort
    unreachable
   end
-  f64.const -nan:0x8000000000000
+  f64.const nan:0x8000000000000
```
Comment (HerrCai0907) [inline tests/compiler/std/math.release.wat:49903]: Change in this file should be reverted. I guess you use mac arm64 as develop machine. I have the same problem.

Comment (yjhmelody) [reply tests/compiler/std/math.release.wat:49903]: Ok

**Comment by yjhmelody** [comment]
Hi @dcodeIO Could this be merged? Though It's not a method of Map in JS, but it's very helpful to reduce twice lookup to once.

**Comment by dcodeIO** [comment]
Generally agree with the observation that it would be good to do something about double lookups. Feels somewhat suboptimal that we'd have to add a non-standard method for this, hence I'm wondering if there are alternatives.

Starting from the underlying issue, that is that `Map#get` traps when a respective entry does not exist, which is already non-standard (since `undefined` doesn't exist), thinkable approaches could be:

* Add a non-standard method `Map#getOrDefault` as proposed, adding additional non-standardness
* Add a second argument to `Map#get`, which is an optional default value (traps if not provided), changing where the non-standardness materializes
* Something else?

Thoughts?

**Comment by dcodeIO** [comment]
Regarding the alternative of modifying `Map#get` with a second parameter, a potential implementation could use `defaultValue: V = changetype<V>(0)`, and only trap when `V` is a non-nullable type and `defaultValue == 0` - the default if `defaultValue` is omitted. This would be a slight breaking change, in that if `V` is nullable, `get` would never trap but return `null` if a key does not exist. One could argue that this breaking change makes it more JS-y in the typical case.

**Comment by yjhmelody** [comment]
Well, should I close this PR?

**Pull request closed without merging**